OpenEdXMobile/src/main/java/org/edx/mobile/base/MainApplication.java

```diff
@@ -132,16 +130,27 @@ private void init() {
         );
     }
 
-    private boolean needVersionUpgrade(Context context) {
-        boolean needVersionUpgrade = false;
-        PrefManager.AppInfoPrefManager pmanager = new PrefManager.AppInfoPrefManager(context);
-        Long previousVersion = pmanager.getAppVersionCode();
-        final int curVersion = BuildConfig.VERSION_CODE;
-        if (previousVersion < curVersion) {
-            needVersionUpgrade = true;
-            pmanager.setAppVersionCode(curVersion);
+    private void checkIfAppVersionUpgraded(Context context) {
+        PrefManager.AppInfoPrefManager prefManager = new PrefManager.AppInfoPrefManager(context);
+        long previousVersionCode = prefManager.getAppVersionCode();
+        final long currVersionCode = (long) BuildConfig.VERSION_CODE;
+        if (previousVersionCode < currVersionCode) {
+            final String previousVersionName = prefManager.getAppVersionName();
+            // Update version code and name in preferences
+            prefManager.setAppVersionCode(currVersionCode);
+            prefManager.setAppVersionName(BuildConfig.VERSION_NAME);
+            // App is updated
+            onAppUpdated(previousVersionCode, currVersionCode, previousVersionName, BuildConfig.VERSION_NAME);
         }
-        return needVersionUpgrade;
+    }
+
+    private void onAppUpdated(final long previousVersionCode, final long currVersionCode,
+                             final String previousVersionName, final String currVersionName) {
+        // Try repair of download data on updating of app version
+        injector.getInstance(IStorage.class).repairDownloadCompletionData();
+        // Fire app updated event
+        EventBus.getDefault().post(new AppUpdatedEvent(previousVersionCode, currVersionCode,
```

Comment:
Instead of `post` we should be using `postSticky` in this case, so that any screen (that hasn't been created yet) can receive this event upon its creation.